main.go

```diff
@@ -0,0 +1,605 @@
+	Type                hexutil.Uint64    `json:"type"`
+	Accesses            *types.AccessList `json:"accessList,omitempty"`
+	ChainID             *hexutil.Big      `json:"chainId,omitempty"`
+	BlobVersionedHashes []common.Hash     `json:"blobVersionedHashes,omitempty"`
+	V                   *hexutil.Big      `json:"v"`
+	R                   *hexutil.Big      `json:"r"`
+	S                   *hexutil.Big      `json:"s"`
+	YParity             *hexutil.Uint64   `json:"yParity,omitempty"`
+}
+
+// eth_getTransactionByHash
+// GetTransactionByHash returns the transaction for the given hash
+func (s *BlockChainAPI) GetTransactionByHash(
+	ctx context.Context,
+	hash common.Hash,
+) (*RPCTransaction, error) {
+	return &RPCTransaction{}, nil
+}
+
+// eth_getTransactionByBlockHashAndIndex
+// GetTransactionByBlockHashAndIndex returns the transaction for the given block hash and index.
+func (s *BlockChainAPI) GetTransactionByBlockHashAndIndex(
+	ctx context.Context,
+	blockHash common.Hash,
+	index hexutil.Uint,
+) *RPCTransaction {
+	return &RPCTransaction{}
+}
+
+// eth_getTransactionByBlockNumberAndIndex
+// GetTransactionByBlockNumberAndIndex returns the transaction for the given block number and index.
+func (s *BlockChainAPI) GetTransactionByBlockNumberAndIndex(
+	ctx context.Context,
+	blockNr rpc.BlockNumber,
+	index hexutil.Uint,
+) *RPCTransaction {
+	return &RPCTransaction{}
+}
+
+// eth_getTransactionReceipt
+// GetTransactionReceipt returns the transaction receipt for the given transaction hash.
+func (s *BlockChainAPI) GetTransactionReceipt(
+	ctx context.Context,
+	hash common.Hash,
+) (map[string]interface{}, error) {
+	return map[string]interface{}{}, nil
+}
+
+// eth_coinbase (return the coinbase for a block)
+// Coinbase is the address that mining rewards will be sent to (alias for Etherbase).
+func (s *BlockChainAPI) Coinbase() (common.Address, error) {
+	return common.Address{}, nil
+}
+
+// eth_getBlockByHash
+// GetBlockByHash returns the requested block. When fullTx is true all transactions in the block are returned in full
+// detail, otherwise only the transaction hash is returned.
+func (s *BlockChainAPI) GetBlockByHash(
+	ctx context.Context,
+	hash common.Hash, fullTx bool,
+) (map[string]interface{}, error) {
+	return map[string]interface{}{}, nil
+}
+
+// eth_getBlockByNumber
+// GetBlockByNumber returns the requested canonical block.
+//   - When blockNr is -1 the chain pending block is returned.
+//   - When blockNr is -2 the chain latest block is returned.
+//   - When blockNr is -3 the chain finalized block is returned.
+//   - When blockNr is -4 the chain safe block is returned.
+//   - When fullTx is true all transactions in the block are returned, otherwise
+//     only the transaction hash is returned.
+func (s *BlockChainAPI) GetBlockByNumber(
+	ctx context.Context,
+	number rpc.BlockNumber,
+	fullTx bool,
+) (map[string]interface{}, error) {
+	return map[string]interface{}{}, nil
```

Comment:
Why returning map[string]any? wouldn't it be better to define a type for it

Reply:
I am assuming that's because the method has to return `nil`. At least that's what I read on the specification page, and the actual implementation:

```go
block, err := s.b.BlockByNumber(ctx, number)
if block != nil && err == nil {
	response, err := s.rpcMarshalBlock(ctx, block, true, fullTx)
	if err == nil && number == rpc.PendingBlockNumber {
		// Pending blocks need to nil out a few fields
		for _, field := range []string{"hash", "nonce", "miner"} {
			response[field] = nil
		}
	}
	return response, err
}
return nil, err
```
And there's a special handling for pending blocks, where they have to set 3 fields to `nil` value.